app/src/main/java/com/example/beetrackerrough/GetLocation.kt

```diff
@@ -15,14 +39,174 @@ import android.view.ViewGroup
+            map?.uiSettings?.isMyLocationButtonEnabled = true
+            map?.uiSettings?.isZoomControlsEnabled = true
+
+            fusedLocationProvider?.lastLocation?.addOnCompleteListener { getLocationTask ->
+                val location = getLocationTask.result
+                if (location != null) {
+                    Log.d(TAG, "User's location $location")
+                    val center = LatLng(location.latitude, location.longitude)
+                    val zoomLevel = 8f
+                    map?.moveCamera(CameraUpdateFactory.newLatLngZoom(center, zoomLevel))
+                    moveMapToUsersLocation = true
+                } else {
+                    showSnackbar(getString(R.string.no_location))
+                }
+            }
+        }
+
+    }
+
+    private fun updateMap() {
+        if (locationPermissionGranted) {
+            if (!moveMapToUsersLocation) {
+                moveMapToUserLocation()
+            }
+        }
+    }
+
+    private fun requestLocationPermission() {
+        //if (isAdded()) {
+        if (ContextCompat.checkSelfPermission(
+                requireActivity(),
+                android.Manifest.permission.ACCESS_FINE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED
+        ) {
+            locationPermissionGranted = true
+            Log.d(TAG, "permission already granted")
+            updateMap()
+            fusedLocationProvider =
+                LocationServices.getFusedLocationProviderClient(requireActivity())
+        } else {
+            val requestLocationPermissionLauncher =
+                registerForActivityResult(ActivityResultContracts.RequestPermission()) { granted ->
+                    if (granted) {
+                        Log.d(TAG, "User granted permission")
+                        locationPermissionGranted = true
+                        fusedLocationProvider =
+                            LocationServices.getFusedLocationProviderClient(requireActivity())
+                    } else {
+                        Log.d(TAG, "User did not grant permission")
+                        locationPermissionGranted = false
+                        showSnackbar(getString(R.string.give_permission))
+                    }
+                    updateMap()
+                }
+            requestLocationPermissionLauncher.launch(android.Manifest.permission.ACCESS_FINE_LOCATION)
+        }
+    }
+
+    private fun showSnackbar(message: String) {
+        Snackbar.make(requireView(), message, Snackbar.LENGTH_LONG).show()
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun addLocation() {
+        if (map == null) { return }
+        if (fusedLocationProvider == null) { return }
+        if (!locationPermissionGranted) {
+            showSnackbar(getString(R.string.no_location))
+            return
+        }
+        fusedLocationProvider?.lastLocation?.addOnCompleteListener(requireActivity()) { locationRequestTask ->
```

Comment:
You should be able to do this in `BeeSightingUserInput` without a separate `Fragment` or opening a map by accessing the `FusedLocationProvider` directly, initializing it in the `onCreateView` method. Something like this (in `BeeSightingUserInput`):
```kotlin
private fun getLocation() {
    if (ActivityCompat.checkSelfPermission(
            requireContext(),
            Manifest.permission.ACCESS_FINE_LOCATION
        ) == PackageManager.PERMISSION_GRANTED
    ) {
        fusedLocationProvider.lastLocation.addOnSuccessListener { location ->
            if (location != null) {
                userLocation = GeoPoint(location.latitude, location.longitude)
            } else {
                Toast.makeText(context, "Unable to fetch location.", Toast.LENGTH_SHORT).show()
            }
        }
    } else {
        requestPermissions(arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), 0)
    }
}

override fun onRequestPermissionsResult(
    requestCode: Int, permissions: Array<out String>, grantResults: IntArray
) {
    if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
        fetchUserLocation()
    } else {
        Toast.makeText(context, "Error: location needed for submission", Toast.LENGTH_SHORT).show()
    }
    super.onRequestPermissionsResult(requestCode, permissions, grantResults)
}
```
And you'd initialize `fusedLocationProvider` like this:
```kotlin
class BeeSightingUserInput : Fragment() {

    /* ... */
    
    private lateinit var fusedLocationClient: FusedLocationProviderClient

    /* ... */

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_bee_sighting_user_input, container, false)

        /* ... */

        fusedLocationClient = LocationServices.getFusedLocationProviderClient(requireContext())
    }
    
    /* ... */
}
```

Reply:
I tried adding your code above to the program and while I could get it to request permissions and send a location to firebase - I started encountering more bugs on the exact location it was sending to firebase.  For example when I was using the extended controls to change my GPS location I couldn't get the userLocation GeoPoint to update properly.  Feel free to take a crack at it if you want.  Otherwise it still appears to be functioning properly with the map being opened in a separate fragment.